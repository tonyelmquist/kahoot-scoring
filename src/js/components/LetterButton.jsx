import React, { Component } from "react";
import { AwesomeButton } from "react-awesome-button";

class LetterButton extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <AwesomeButton
        className="letter-button"
        action={() => {
          this.props.incrementTally(this.props.item.name);
        }}
      >
        {this.props.item.name}
      </AwesomeButton>
    );
  }
}

export default LetterButton;
