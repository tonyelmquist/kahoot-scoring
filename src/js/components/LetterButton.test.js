import React from 'react';
import enzymeSetup from '../utils/enzymeSetup';
import {shallow} from 'enzyme';
import LetterButton from './LetterButton';


describe('LetterButton tests', () => {
    const item= {
          name: "B",
          value: 30,
          bonusInterval: 2,
          bonusValue: 90,
          tally: 0
        }

    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<LetterButton item={item}/>);
    });

    test('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
        console.log(wrapper);
    });

});