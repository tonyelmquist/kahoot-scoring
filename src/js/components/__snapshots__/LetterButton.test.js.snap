// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LetterButton tests renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LetterButton
    item={
      Object {
        "bonusInterval": 2,
        "bonusValue": 90,
        "name": "B",
        "tally": 0,
        "value": 30,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "action": [Function],
      "active": false,
      "blocked": false,
      "bubbles": false,
      "children": "B",
      "className": "letter-button",
      "cssModule": null,
      "disabled": false,
      "element": null,
      "href": null,
      "moveEvents": true,
      "placeholder": false,
      "rootElement": "aws-btn",
      "size": null,
      "style": Object {},
      "target": null,
      "title": null,
      "to": null,
      "type": "primary",
      "visible": true,
    },
    "ref": null,
    "rendered": "B",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "action": [Function],
        "active": false,
        "blocked": false,
        "bubbles": false,
        "children": "B",
        "className": "letter-button",
        "cssModule": null,
        "disabled": false,
        "element": null,
        "href": null,
        "moveEvents": true,
        "placeholder": false,
        "rootElement": "aws-btn",
        "size": null,
        "style": Object {},
        "target": null,
        "title": null,
        "to": null,
        "type": "primary",
        "visible": true,
      },
      "ref": null,
      "rendered": "B",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
